### ISSUE TITLE
Load all of GHC in GHCi

### ISSUE THREAD

**Issue by angerman** [snowleopard/hadrian#551]
As raised by @nomeata on [ghc-devs](https://mail.haskell.org/pipermail/ghc-devs/2018-April/015567.html), it would be nice if Hadrian had a rule to load all of GHC into GHCi.

**Comment by mpickering** [comment]
See https://ghc.haskell.org/trac/ghc/wiki/Building/InGhci

**Comment by snowleopard** [comment]
@angerman By "rule" do you mean a build rule? Could you clarify how it would look like?

Say, we call it `repl` -- what would `hadrian repl` need to build/do?

**Comment by mpickering** [comment]
I think that ideally you would also need to build the in-tree `ghc-boot-th` package. I remember some problems with this. 

In my mind, calling `hadrian repl` will load a GHCi session with all modules of GHC loaded.

**Comment by mgsloan** [comment]
FYI I am looking into adding this to hadrian.  Whatever I come up with will likely depend on D4904 being merged.

**Comment by alpmestan** [comment]
Feel free to use this issue for asking questions about problems encountered while trying to implement this, if any. And thanks, because this should be quite useful. I'll make sure to documet the ghci command [here](https://ghc.haskell.org/trac/ghc/wiki/Building/Hadrian/QuickStart) once you're done.

**Comment by mgsloan** [comment]
Not going to get around to adding this to hadrian tonight, sorry!  Might revisit.  I couldn't get hadrian to work - first I ran into https://github.com/snowleopard/hadrian/issues/631 - but then I realized I was using the ghc submodule version of hadrian.  Then I ran into https://github.com/snowleopard/hadrian/issues/634 , so I switched back to standrard old `make`, but then I had issues with that that never occurred before (some business about `unlit/fs.c`).  Eventually I tried boot + configure and that seems to resolve it.  Since the build is working again, just going to keep at it this way for the time being
